### ISSUE TITLE
NA in seurat metadata

### ISSUE THREAD

**Issue by ShaktiprasadM** [satijalab/seurat#6721]
I am using GSE70630 from GEO for analysis. I first converted the txt file into sparse matrix. After creating seurat object, I am seeing there are "NA" values in metadata. How to remove it??

![image](https://user-images.githubusercontent.com/119098992/204077335-c1996c6f-c78f-4381-9f2a-79ab9a6105a7.png)

**Comment by mhkowalski** [comment]
Could you please clarify what you mean by removing the NAs? The presence of an NA indicates that a particular piece of metadata was not available for that cell. I would not recommend replacing them with 0's, for instance, because this could affect any downstream calculations you do with the metadata.

**Comment by ShaktiprasadM** [comment]
> Could you please clarify what you mean by removing the NAs? The presence of an NA indicates that a particular piece of metadata was not available for that cell. I would not recommend replacing them with 0's, for instance, because this could affect any downstream calculations you do with the metadata.

These places actually have 0 or value in decimal in the countdata. But after making Seurat object, they have become NA. How to solve this issue??

**Comment by mhkowalski** [comment]
In general, when trying to meta data from an external file, I would first make the meta data into a data.frame (which should coerce any "." into 0's). You can then check if there are any NAs in this data.frame, and replace them with 0's if you would like (many ways to do this that are outside of the scope of Seurat, I would check stackoverflow if you need assistance). 

Finally, you can add this data.frame into the Seurat object using the `AddMetaData` function.

**Comment by changostraw** [comment]
I have a similar issue. I have a dataframe with metadata where there are no NAs or empty cells.

Barcode	Demuxlet_DropletType	Demuxlet_Individual_Assignment	scds_score	scds_DropletType
AAACCCAAGACATATG-1	unassigned	unassigned	0.12566983554583	singlet
AAACCCAAGAGTTGCG-1	singlet	325_GT12938	0.160387636698164	singlet
AAACCCAAGCCTCACG-1	singlet	309_GT12139	0.151774600201326	singlet
AAACCCACAACTCCAA-1	singlet	309_GT12139	0.337062172839279	singlet
AAACCCACAATGTCTG-1	singlet	366_GT15314	0.141561653238646	singlet
AAACCCACATCGCTGG-1	singlet	309_GT12139	0.43772493142067	singlet
AAACCCAGTAACTTCG-1	singlet	366_GT15314	0.914639250049845	doublet
AAACCCAGTGAGACCA-1	doublet	doublet	0.377536071516808	singlet
AAACCCAGTTCGTACA-1	singlet	309_GT12139	0.443144124318555	singlet
AAACCCATCAAGAGTA-1	unassigned	unassigned	0.0538829704628992	singlet
AAACCCATCCCGTTGT-1	singlet	309_GT12139	0.470997677356941	singlet
AAACGAAAGACAACAT-1	unassigned	unassigned	0.418480805126437	singlet
AAACGAAAGCCTCACG-1	singlet	366_GT15314	1.11414881367849	doublet
AAACGAAAGCTCGTGC-1	singlet	366_GT15314	0.507740265005919	singlet
AAACGAAAGTCTCTGA-1	singlet	325_GT12938	0.35888661815478	singlet

I used:
`CECA_CP <- AddMetaData(object = CECA_CP, metadata = cecum_cp_imputed_combined_results, col.name = NULL)`

The colnames are added, but for every cell I get an NA. 

                     orig.ident nCount_RNA nFeature_RNA Barcode Demuxlet_DropletType Demuxlet_Individual_Assignment scds_score
AAACCCACAAGATTGA-1 SeuratProject        512          414    <NA>                 <NA>                           <NA>         NA
AAACCCACAGAGGGTT-1 SeuratProject        335          235    <NA>                 <NA>                           <NA>         NA
AAACCCAGTGCCCAGT-1 SeuratProject       1358          272    <NA>                 <NA>                           <NA>         NA
AAACCCATCACTGTCC-1 SeuratProject        548          506    <NA>                 <NA>                           <NA>         NA
AAACCCATCATTGCCC-1 SeuratProject        351          234    <NA>                 <NA>                           <NA>         NA
AAACCCATCGATACGT-1 SeuratProject        312          211    <NA>                 <NA>                           <NA>         NA
AAACCCATCGCAGAGA-1 SeuratProject        374          314    <NA>                 <NA>                           <NA>         NA
AAACGAAAGACTGAGC-1 SeuratProject        411          347    <NA>                 <NA>                           <NA>         NA
AAACGAAAGATGCTAA-1 SeuratProject        317          295    <NA>                 <NA>                           <NA>         NA
AAACGAAAGCTCGGCT-1 SeuratProject        320          291    <NA>                 <NA>                           <NA>         NA
AAACGAAAGGTAAGAG-1 SeuratProject        859          697    <NA>                 <NA>                           <NA>         NA
AAACGAAAGGTCCAGA-1 SeuratProject        544          501    <NA>                 <NA>                           <NA>         NA
AAACGAAAGTCTGTAC-1 SeuratProject        371          352    <NA>                 <NA>                           <NA>         NA
AAACGAACAACGGGTA-1 SeuratProject        364          322    <NA>                 <NA>                           <NA>         NA
AAACGAACAATACGAA-1 SeuratProject       2597         1649    <NA>                 <NA>                           <NA>         NA
AAACGAAGTCTCGGGT-1 SeuratProject        878          409    <NA>                 <NA>                           <NA>         NA
AAACGAAGTTTGTTGG-1 SeuratProject        377          248    <NA>                 <NA>                           <NA>         NA
AAACGAATCAGCAGAG-1 SeuratProject        328          271    <NA>                 <NA>                           <NA>         NA
AAACGAATCCTATTGT-1 SeuratProject        333          298    <NA>                 <NA>                           <NA>         NA
AAACGCTCAAACCATC-1 SeuratProject        704          272    <NA>                 <NA>                           <NA>         NA
AAACGCTCAAATGGCG-1 SeuratProject       2326         1524    <NA>                 <NA>                           <NA>         NA
AAACGCTCAGGTAGTG-1 SeuratProject        386          376    <NA>                 <NA>                           <NA>         NA
AAAGAACAGAAACCCG-1 SeuratProject        307          298    <NA>                 <NA>                           <NA>         NA
AAAGAACCAACCCTAA-1 SeuratProject        456          442    <NA>                 <NA>                           <NA>         NA
AAAGAACCATACATCG-1 SeuratProject        325          255    <NA>                 <NA>                           <NA>         NA
AAAGAACCATATCTGG-1 SeuratProject        345          285    <NA>                 <NA>                           <NA>         NA

I also tried adding one column at a time:
`CECA_CP$demux <- cecum_cp_imputed_combined_results$Demuxlet_DropletType`

but get an error:
Error: Cannot add more or fewer cell meta.data information without values being named with cell names

The Seurat object and the data.frame have the same number and label of cell names. 

I must be missing something? thanks!

**Comment by changostraw** [comment]
nevermind - I realized I was missing something obvious - my data.frame didn't have the barcodes (cell names) as the rownames. Once I set the rownames to be the barcodes `row.names(cecum_cp_imputed_combined_results) <- cecum_cp_imputed_combined_results$Barcode` the addMetaData command above worked.

**Comment by yojetsharma** [comment]
> nevermind - I realized I was missing something obvious - my data.frame didn't have the barcodes (cell names) as the rownames. Once I set the rownames to be the barcodes `row.names(cecum_cp_imputed_combined_results) <- cecum_cp_imputed_combined_results$Barcode` the addMetaData command above worked.

I am trying to create a seurat object using the data set here: http://solo.bmap.ucla.edu/shiny/webapp/ 
I loaded the dataset as mentioned on the website and created a matrix. Following which, I created a seurat object. Then, I read the meta data (provided by the authors on the website) using read.csv function and tried adding the metadata into the seurat object. But it results in NAs. 
![Screenshot 2023-03-07 at 23 19 59](https://user-images.githubusercontent.com/79513243/223520229-ffb5f0e1-7faf-4c39-bb9c-f749fa86612a.png)
![Screenshot 2023-03-07 at 23 19 39](https://user-images.githubusercontent.com/79513243/223520312-2e363c58-5f5f-46d5-ae60-46a1d6e057dd.png)
I am guessing that the seurat object that I created has a column that consists of barcodes has its name missing from it but in my metadata the barcode column is named as Cell. How can I fix this?
